src/postgres.c

```diff
@@ -297,15 +369,39 @@ void construct_dump_path(char *path)
 	strcat(path, "/backup.dump");
 }
 
-void read_buffer_pipe(int *pipefd)
+int read_buffer_pipe(int *pipefd)
 {
 	char buffer[4096] = { 0 };
 	int nbytes = read(pipefd[READ_END], buffer, sizeof(buffer));
+	int no_of_new_lines = 0;
+
+	// We keep the sum of the bytes so we can correctly reallocate memory if needed.
+	int sum_of_bytes = nbytes;
+
+	if (nbytes < 0) {
+		return -1;
+	}
+
 	while (nbytes > 0) {
 		pr_info("%s", buffer);
+		format_buffer(buffer);
+
+		if (sum_of_bytes > 4096) {
+			email_body = realloc(email_body,
+					     sum_of_bytes * sizeof(char) + 1);
+		}
```

Comment:
I propose that we `realloc` in all-cases, i.e. once we know how many bytes the string is ending up being, just `realloc` to that and save space.

Reply:
Wouldn't we have a lot of overhead added by calling realloc again and again? If i get the string from the buffer and realloc I might have to all it 20+ times.

Reply:
Ideally, we'll keep a counter in a global scope, and `realloc` to that **only if we didn't `realloc`  in the first place because of smaller buffer size.**.

Reply:
And if we do have to realloc earlier, should I just realloc to the counter when im done reading?

Reply:
No, a `bool realloced;` is enough.

```suggestion
		if (sum_of_bytes > 4096) {
			email_body = realloc(email_body,
					     sum_of_bytes * sizeof(char) + 1);
			realloced = true;
		}
```